felix/bpf-gpl/tc.c

```diff
@@ -407,6 +407,12 @@ static CALI_BPF_INLINE void calico_tc_process_ct_lookup(struct cali_tc_ctx *ctx)
 	} else {
 		ctx->state->post_nat_ip_dst = ctx->state->ip_dst;
 		ctx->state->post_nat_dport = ctx->state->dport;
+		if (nat_res == NAT_EXCLUDE) {
+			/* We want such packets to go through the host namespace. The main
+			 * usecase of this is node-local-dns.
+			 */
+			ctx->state->flags |= CALI_ST_SKIP_FIB;
```

Comment:
Do we need to mark the conntrack entry too?  Otherwise only the first packet will skip the FIB.

Reply:
Correct, that is done where we create the ct entry here. That is an already established process. https://github.com/tomastigera/project-calico-calico/blob/tomas-bpf-node-local-dns/felix/bpf-gpl/tc.c#L1231-L1232